src/Features/Core/Portable/UseThrowExpression/AbstractUseThrowExpressionDiagnosticAnalyzer.cs

```diff
@@ -77,15 +77,21 @@ private void AnalyzeOperation(OperationAnalysisContext context, INamedTypeSymbol
 
             var cancellationToken = context.CancellationToken;
 
-            var throwOperation = (IThrowStatement)context.Operation;
-            var throwStatement = throwOperation.Syntax;
+            var throwExpression = (IThrowExpression)context.Operation;
+            var throwStatementOperation = throwExpression.Parent;
+            if (throwStatementOperation.Kind != OperationKind.ExpressionStatement)
+            {
+                return;
+            }
+
+            var throwStatement = throwExpression.Syntax;
```

Comment:
It doesn't look fixed to me, though I am not sure if there is anything wrong with this code. To clarify, I was commenting on the fact that statement syntax was pulled from expression. Shouldn't it be pulled from the statement instead?